src/trace/implementations/mod.rs

```diff
@@ -320,6 +363,109 @@ impl BatchContainer for OffsetList {
     }
 }
 
+/// Behavior to split an update into principal components.
+pub trait BuilderInput<L: Layout> {
```

Comment:
We discussed whether this might work with a `BuilderInput<'a, L>`. We didn't conclude that it could or couldn't, but if it ended up being `C::Item<'a>: BuilderInput<'a, L>` that might be clearer than the alternative.

Reply:
Another thought we had: Perhaps `BuilderInput<L>: Container`, and rather than introduce an `Item<'a>` here and equate it to `C::Item<'a>`, the signatures could just use `C::Item<'a>`.